Sources/TuistCache/Cache/FileDownloader.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import RxSwift
+import TSCBasic
+
+enum FileDownloaderError: LocalizedError {
+    case noLocalURL
+    case invalidResponse
+    case urlSessionError(Error)
+    case moveFileError(Error)
+
+    var errorDescription: String? {
+        switch self {
+        case .noLocalURL:
+            return "Didn't receive a valid local URL"
+        case .invalidResponse:
+            return "Invalid URL response"
+        case let .urlSessionError(error):
+            if let error = error as? LocalizedError {
+                return error.localizedDescription
+            } else {
+                return "Error while parsing the network response."
```

Comment:
It's not really while parsing. URLSession failed when we told it to run a request.

Reply:
Addressed